packages/plugin-ext/src/plugin/command-registry.ts

```diff
@@ -86,3 +96,75 @@ export class CommandRegistryImpl implements CommandRegistryExt {
         }
     }
 }
+
+export class CommandsConverter {
+
+    private readonly delegatingCommandId: string;
+
+    private cacheId = 0;
+    private cache = new Map<number, theia.Command>();
+
+    constructor(private readonly commands: CommandRegistryImpl) {
+        this.delegatingCommandId = `_internal_command_delegation_${Date.now()}`;
+        this.commands.registerHandler(this.delegatingCommandId, this.executeConvertedCommand);
+    }
+
+    toInternal(command: theia.Command | undefined): Command | undefined {
+        if (!command || !command.label) {
+            return undefined;
+        }
+
+        const result: Command = {
+            id: command.id,
+            title: command.label
+        };
+
+        if (command.id && !this.isFalsyOrEmpty(command.arguments)) {
+            const id = this.cacheId++;
+            ObjectIdentifier.mixin(result, id);
+            this.cache.set(id, command);
+
+            result.id = this.delegatingCommandId;
+            result.arguments = [id];
+        }
+
+        if (command.tooltip) {
+            result.tooltip = command.tooltip;
+        }
+
+        return result;
+    }
+
+    fromInternal(command: Command | undefined): theia.Command | undefined {
+        if (!command) {
+            return undefined;
+        }
+
+        const id = ObjectIdentifier.of(command);
+        if (typeof id === 'number') {
+            return this.cache.get(id);
+
+        } else {
+            return {
+                id: command.id,
+                label: command.title,
+                arguments: command.arguments
+            };
+        }
+    }
+
+    private executeConvertedCommand(...args: any[]): PromiseLike<any> {
+        const actualCmd = this.cache.get(args[0]);
+        if (!actualCmd) {
+            return Promise.resolve(undefined);
```

Comment:
curious, could not be just `return undefined` ?

Reply:
no